### PR TITLE
Ks 2022 09 support phase

### PR DESCRIPTION
**Pull request by Rineee**
@fuzzylogic2000 Maybe you already want to have a look, especially the rules (left it all in mb for now), but there a still some todos:

- [x] remove ratings from list view (-> add supoort there?)
- [ ] remove ratings from map pop up (-> add support there?)
- [x] change filter wording for support phase

Not even sure about the 'adding support' part of the first two here, is that included in the story? Not explicitly at least :)

### CODE REVIEW

File: meinberlin/apps/budgeting/rules.py [hunk 1]
```diff
@@ -20,9 +21,13 @@
 
 rules.add_perm(
     'meinberlin_budgeting.rate_proposal',
-    module_predicates.is_allowed_rate_item
+    module_predicates.is_allowed_rate_item | is_allowed_support_item
```
Comment (fuzzylogic2000) [inline meinberlin/apps/budgeting/rules.py:24]: I think you already explained it, but could you do that again please?
Why do we need `is_allowed_support` here, but not `is_allowed_rate` in the support proposal perm?

File: meinberlin/apps/budgeting/assets/SupportBox.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react'
```
Comment (fuzzylogic2000) [inline meinberlin/apps/budgeting/assets/SupportBox.jsx:1]: This is just the RateBox from a4, but changed according to supporting, right?

File: meinberlin/apps/budgeting/templates/meinberlin_budgeting/proposal_detail.html [hunk 3]
```diff
@@ -13,6 +14,17 @@
     </span>
 {% endblock %}
 
+{% block ratings %}
+    {% if module.blueprint_type == 'PB3' and proposal|has_feature:"support" %}
```
Comment (fuzzylogic2000) [inline meinberlin/apps/budgeting/templates/meinberlin_budgeting/proposal_detail.html:18]: Ah, yes, nice!

**Review by fuzzylogic2000 — COMMENTED** [review]
Looks good! Even though there is still the stuff we need to think about and handle, it looks like a good way to me

File: meinberlin/apps/budgeting/rules.py (hunk 1 above)
Comment (Rineee) [reply meinberlin/apps/budgeting/rules.py:24]: @fuzzylogic2000 We need it here for the rating api, that we are using and that checks for rate_proposal on POST.

File: meinberlin/apps/budgeting/assets/SupportBox.jsx (hunk 2 above)
Comment (Rineee) [reply meinberlin/apps/budgeting/assets/SupportBox.jsx:1]: Well I rewrote it bit to a functional component and threw out some stuff that we dont need here, but yes, the base is the rating box :)

**Review by fuzzylogic2000 — COMMENTED** [review]
@Rineee I thought handling as much as possible in the python code is a good idea? Then it's also easy to change when the logic when we show what is changed. (like in another phase as well or between certain ones or whatever.) What do you think?

**Comment by Rineee** [comment]
> @Rineee I thought handling as much as possible in the python code is a good idea? Then it's also easy to change when the logic when we show what is changed. (like in another phase as well or between certain ones or whatever.) What do you thin

@fuzzylogic2000 Really cool, I like it this way. I also tried to fix the frontend tests, hope the right way :see_no_evil: I would say we could merge now and do the pop up part once it is fixed?

**Comment by fuzzylogic2000** [comment]
> > @Rineee I thought handling as much as possible in the python code is a good idea? Then it's also easy to change when the logic when we show what is changed. (like in another phase as well or between certain ones or whatever.) What do you thin
> 
> @fuzzylogic2000 Really cool, I like it this way. I also tried to fix the frontend tests, hope the right way see_no_evil I would say we could merge now and do the pop up part once it is fixed?

@Rineee Yes, I think merging now and fixing the map popups in the second map is a good idea! But I think one of @phillimorland @khamui and @goapunk should also approve and do the merging. :)

File: meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx [hunk 4]
```diff
@@ -2,9 +2,16 @@ import React from 'react'
 import { render, screen } from '@testing-library/react'
 import { ListItemStats } from '../ListItemStats'
 
-test('displaying all 3 stats', () => {
+test('rating displaying all 3 stats', () => {
+  const permissions = {
+    view_support_count: false,
```
[the remark this answers is not in the document]
Comment (fuzzylogic2000) [reply meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx:7]: Hmm, as we never show all stats on the real list, I don't know. Maybe there should be two tests, one showing the supports and the other one the rating?

**Comment by phillimorland** [comment]
> > @fuzzylogic2000 @Rineee code looks good and tests pass, just one quick question about the 3 stats. Also should the default phase descriptions in dashboard be updated? and also how do I give myself support permission? couldn't figure that out in dashboard
> 
> @phillimorland Oh, sorry, should have said! The supporting only works with a new 3 phase Bürgerhaushalt as the old one still has the old phase.

@fuzzylogic2000 cheers, looks good, only thing is the support and the moderation buttons are on 2 separate lines but that can be fixed later so LGTM from me :)

File: meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (fuzzylogic2000) [reply meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx:7]: oh no, overread that. Yes!

[the remark this answers is not in the document]
Comment (Rineee) [reply meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx:7]: @phillimorland @fuzzylogic2000 So this test is testing if its showing all 3 stats (pos, neg, comments) when there is a rating phase and the other one below is testing that only 2 are shown (pos, comments) when there is support.. does that make sense?

[the remark this answers is not in the document]
Comment (Rineee) [reply meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx:7]: So it is true to show all 3, see lines 20-25.

**Comment by Rineee** [comment]
> > > @fuzzylogic2000 @Rineee code looks good and tests pass, just one quick question about the 3 stats. Also should the default phase descriptions in dashboard be updated? and also how do I give myself support permission? couldn't figure that out in dashboard
> > 
> > 
> > @phillimorland Oh, sorry, should have said! The supporting only works with a new 3 phase Bürgerhaushalt as the old one still has the old phase.
> 
> @fuzzylogic2000 cheers, looks good, only thing is the support and the moderation buttons are on 2 separate lines but that can be fixed later so LGTM from me :)

Oh yes, I saw that and then ignored it :see_no_evil: Good you pointed it out again @phillimorland

File: meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (fuzzylogic2000) [reply meinberlin/apps/budgeting/assets/__tests__/ListItemStats.jest.jsx:7]: Yes, cool. Just delete the task again. :)

**Pull request merged**